utils/navMenus.js

```diff
@@ -0,0 +1,90 @@
+import Search from "../configuration/config";
+
+import key from "../credentials";
+
+import { menuMore, icons, menus } from "./menuObjects";
+
+const data = {};
+
+/**
+ * Creates an object for each sub-menu that contains a title property and a subArticles property
+ * that contains an object that will generate a number of subArticles for each sub-menu, this number being 5.
+ *
+ *
+ * @param {int} numberOfSubMenus Takes the length of the array of the sub menus to now how many sub menus are.
+ * @param {array} arrayOfSubMenus An array for all the menus of the navbar
+ * @param {string} query A string that will be used to search the articles for each submenu.
+ * @param {int} current Controls the number of sub-menus that each menu should have(5)
+ * @param {int} next Controls the number of sub-menus that each menu should have(5)
+ * @returns {Array} An array of all the subMenus and Articles that have each sub-menu
+ */
+async function createSubMenu(
+  numberOfSubMenus,
+  arrayOfSubMenus,
+  query,
+  current = 0,
+  next = 5
+) {
+  const subArticlesArray = [];
+
+  for (let index = 0; index < numberOfSubMenus; index += 1) {
+    const searchResults = await query.getResults(arrayOfSubMenus[index]);
+    const subArticle = {
+      title: arrayOfSubMenus[index],
+      subArticles: await Promise.all(
+        searchResults.slice(current, next).map(elem => {
+          return {
+            title: elem.title,
+            url: elem.urlToImage,
+            source: elem.source,
+            author: elem.author,
+            link: elem.url
+          };
+        })
+      )
+    };
+    subArticlesArray.push(subArticle);
+    current += 5;
+    next += 5;
+  }
+  return subArticlesArray;
+}
+
+/**
+ * Takes the menus with type menu and type more concat them in a list creates and object and fill them with the createSubMenu function with the data that it fetches from the news api
+ */
+const searchControl = async () => {
+  data.search = new Search(key.firstKey);
+  const current = 0;
+  const next = 5;
+  const menuNames = menus;
+  const allMenus = menuNames.concat(menuMore);
+
+  try {
+    const values = allMenus.map(async menu => {
+      if (menu.type === "subMenu") {
+        return {
+          name: menu.name,
+          subMenu: await createSubMenu(
+            menu.subMenus.length,
+            menu.subMenus,
+            data.search,
```

Comment:
I don't see why you need the `data` object in the first place. You could just call `new Search(key.firstKey)` here or, if you prefer, do `const search = new Search(key.firstKey)` and pass `search` here.